src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -1202,9 +1203,17 @@ private void stripHopByHopHeaders(HttpHeaders headers) {
      * @param httpRequest the HttpRequest that is resulting in the Bad Gateway response
      * @return true if the connection will be kept open, or false if it will be disconnected
      */
-    private boolean writeBadGateway(HttpRequest httpRequest) {
+    private boolean writeBadGateway(HttpRequest httpRequest, Throwable cause) {
         String body = "Bad Gateway: " + httpRequest.getUri();
-        FullHttpResponse response = ProxyUtils.createFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_GATEWAY, body);
+        HttpResponseStatus status = HttpResponseStatus.BAD_GATEWAY;
+
+        if (cause instanceof SSLHandshakeException) {
```

Comment:
instead of doing this can we implement a method which can be overridden? this way implementors would be able to customize the message based on their own logic. if the `writeBadGateway` was protected for instance we could override it.